### PR TITLE
Added toString implementation to PowerSource

### PR DESCRIPTION
**Pull request by colinbobolin**
PowerSource toString method as specified in issue #942

### CODE REVIEW

File: oshi-core/src/main/java/oshi/hardware/common/AbstractPowerSource.java [hunk 1]
```diff
@@ -71,4 +71,10 @@ public double getRemainingCapacity() {
     public double getTimeRemaining() {
         return this.timeRemaining;
     }
+
+    /** {@inheritDoc} */
+    @Override
+    public String toString() {
+        return String.format("%n %s @ %.1f%%", this.getName(), this.getRemainingCapacity() * 100d);
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/common/AbstractPowerSource.java]: Let's not include the `%n ` header here.

File: oshi-core/src/test/java/oshi/SystemInfoTest.java [hunk 2]
```diff
@@ -278,8 +278,7 @@ private static void printPowerSources(PowerSource[] powerSources) {
             }
         }
         for (PowerSource powerSource : powerSources) {
-            sb.append(
-                    String.format("%n %s @ %.1f%%", powerSource.getName(), powerSource.getRemainingCapacity() * 100d));
+            sb.append(powerSource.toString());
```
Comment (dbwiddis) [inline oshi-core/src/test/java/oshi/SystemInfoTest.java]: Append the `"\n "` here.

File: oshi-core/src/main/java/oshi/hardware/common/AbstractPowerSource.java (hunk 1 above)
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/hardware/common/AbstractPowerSource.java]: Also while it wasn't in the original `SystemInfoTest` we should add `getTimeRemaining()`.  If positive, seconds remaining (would be awesome to format as something like `1h 20m 15s` or even omit seconds and do `1:20`). If negative, -1.0 (calculating) or -2.0 (unlimited)... careful with floating point equality here too, you probably want to do "<1.5" and "<0" cases.

**Comment by dbwiddis** [comment]
Thanks!  I'll do some formatting tweaks and changelog, and then merge this.  Also the modulo with the double worries me just enough that I'll do the int cast before the modulo.

**Comment by colinbobolin** [comment]
Thanks for the review, I'll keep a closer eye on my type casting in the future. Happy I could contribute.

**Comment by dbwiddis** [comment]
Heh, actually it looks like I did it exactly that way in SystemInfoTest but only for the first element of the array. I'm nixing that and leaving the casting. It works, I'm always iffy about floating point math!  I did change the "Unlimited" to "Charging" though... thanks again for your submission.

**Comment by dbwiddis** [comment]
Merged! Thanks for contributing to open source!

**Pull request merged**